controllers/controller.go

```diff
@@ -194,6 +194,10 @@ func (r *EtcdadmClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			etcdCluster.Status.Ready = false
 		}
 
+		if conditions.IsFalse(etcdCluster, etcdv1.EtcdClusterResizeCompleted) {
+			etcdCluster.Status.Ready = false
+		}
```

Comment:
what if the user tries to do upgrade + resize at once? how does this behave?

Reply:
It doesn't make sense to do this in "status.go" mainly due to a couple reasons.

updateStatus in status.go returns early with different scenarios, such as when resize is incomplete, when machines aren't healthy yet, etc. In addition, status.go handles updating the `resizeCompleted` condition in the case that `ownedMachines != desiredReplicas`.

We want this check/update to be done every reconcile loop to make sure the status is properly up to date, and we can effectively do that with how status.go is written. 
We also follow the same flow with another condition in the line above as well.
`if conditions.IsFalse(etcdCluster, etcdv1.EtcdMachinesSpecUpToDateCondition)`

Reply:
> It doesn't make sense to do this in "status.go" mainly due to a couple reasons.

Maybe i didn't understand correctly. Is your reason to do it here that it doesn't make sense to do in status.go or that it is easier to do it here?

Because in terms of code organization, doing it in status.go seems to make more sense.

Reply:
It doesn't make sense to do it in status.go because of the way status.go is written

Reply:
can we remove this now that we have it in status.go?